nft_ingester/src/backfiller.rs

```diff
@@ -0,0 +1,702 @@
+            MAX_FAILURE_DELAY_MS
+        } else {
+            self.failure_delay
+        };
+    }
+
+    fn reset_delay(&mut self) {
+        self.failure_delay = INITIAL_FAILURE_DELAY;
+    }
+
+    async fn get_trees_to_backfill(&self) -> Result<Vec<BackfillTree>, DbErr> {
+        // Get trees with the `force_chk` flag set.
+        let force_chk_trees = UniqueTree::find_by_statement(Statement::from_sql_and_values(
+            DbBackend::Postgres,
+            "SELECT DISTINCT backfill_items.tree FROM backfill_items\n\
+            WHERE backfill_items.force_chk = TRUE",
+            vec![],
+        ))
+        .all(&self.db)
+        .await?;
+
+        // Convert this Vec of `UniqueTree` to a Vec of `BackfillTree` (which contain extra info).
+        let mut trees: Vec<BackfillTree> = force_chk_trees
+            .into_iter()
+            .map(|tree| BackfillTree::new(tree, true))
+            .collect();
+
+        // Get trees with multiple rows from `backfill_items` table.
+        let multi_row_trees = UniqueTree::find_by_statement(Statement::from_sql_and_values(
+            DbBackend::Postgres,
+            "SELECT backfill_items.tree FROM backfill_items\n\
+            GROUP BY backfill_items.tree\n\
+            HAVING COUNT(*) > 1",
+            vec![],
+        ))
+        .all(&self.db)
+        .await?;
+
+        // Convert this Vec of `UniqueTree` to a Vec of `BackfillTree` (which contain extra info).
+        let mut multi_row_trees: Vec<BackfillTree> = multi_row_trees
+            .into_iter()
+            .map(|tree| BackfillTree::new(tree, false))
+            .collect();
+
+        trees.append(&mut multi_row_trees);
+
+        Ok(trees)
+    }
+
+    async fn backfill_tree_from_seq_1(&self, tree: &[u8]) -> Result<Option<i64>, IngesterError> {
+        //TODO implement gap filler that gap fills from sequence number 1.
+        //For now just return the max sequence number.
+        self.get_max_seq(tree).await.map_err(Into::into)
+    }
+
+    async fn get_max_seq(&self, tree: &[u8]) -> Result<Option<i64>, DbErr> {
+        //TODO Find better, simpler query for this.
+        let rows = backfill_items::Entity::find()
+            .filter(backfill_items::Column::Tree.eq(tree))
+            .filter(
+                Condition::any().add(
+                    backfill_items::Column::Seq.in_subquery(
+                        Query::select()
+                            .expr(backfill_items::Column::Seq.max())
+                            .from(backfill_items::Entity)
+                            .and_where(backfill_items::Column::Tree.eq(tree))
+                            .to_owned(),
+                    ),
+                ),
+            )
+            .all(&self.db)
+            .await?;
+
+        Ok(rows.last().map(|row| row.seq))
+    }
+
+    async fn clear_force_chk_flag(&self, tree: &[u8]) -> Result<UpdateResult, DbErr> {
+        backfill_items::Entity::update_many()
```

Comment:
I think I am favoring SQL statements that assume we might have duplicates of things, and I try to always deal with the duplicates and delete the extra stuff, clean as we go.

But also, I don't mean to "update many" here specifically, I think this is just the syntax so that I generate one call:
`UPDATE backfill_items SET backfill_items.force_chk = FALSE where backfill_items.tree = {tree}`

I think if I used the `Model`/`ActiveModel` with plain `update` I think it results in the background like a `SELECT *` followed by an `UPDATE`.